engine/script/src/http_service.cpp

```diff
@@ -395,28 +393,15 @@ namespace dmHttpService
             worker->m_Service = service;
             worker->m_CacheFlusher = i == 0 && worker->m_Service->m_HttpCache != 0;
             worker->m_Run = true;
-            worker->m_DNSChannel = 0;
             service->m_Workers.Push(worker);
 
-// Do not create a c-ares DNS channel on iOS
-// If we have no DNS channel we will fall back to POSIX getaddrinfo
-// c-ares DNS resolution will trigger a Network Security popup on most
-// residential WiFi setups since the WiFi router will act as a DNS and forward
-// DNS queries to the DNS of the ISP.
-// This will trigger the Network Security popup since c-ares communicates with
-// the router over the local network.
-// getaddrinfo on the other hand seems to be whitelisted in iOS.
-#if !defined(DM_PLATFORM_IOS)
-            dmDNS::NewChannel(&worker->m_DNSChannel);
-#endif
-
             dmThread::Thread t = dmThread::New(&Loop, THREAD_STACK_SIZE, worker, "http");
-// Detach the thread on iOS
-// DNS lookups on iOS are using getaddrinfo which may block for an undefined
-// amount of time. We do not wish to wait for the thread during shutdown
-#if defined(DM_PLATFORM_IOS)
+
+            // Detach the thread
+            // DNS lookups using dmSocket::GetHostByName are using getaddrinfo which may block for an undefined
+            // amount of time. We do not wish to wait for the thread during shutdown
```

Comment:
Now we _always_ detach the thread.

@britzl Do you remember why we detach directly after creation, and not at shutdown?

Reply:
No. The documentation doesn't really say that you should use detach() at a specific point in a threads lifetime. You can create threads in a detached state or you can wait until shutdown, when you normally would call join().